pulsar-broker/src/main/java/org/apache/pulsar/broker/web/PulsarWebResource.java

```diff
@@ -501,91 +509,127 @@ protected void validateDestinationOwnership(DestinationName fqdn, boolean author
         }
     }
 
-    protected void validateReplicationSettingsOnNamespace(String property, String cluster, String namespace) {
-        NamespaceName namespaceName = new NamespaceName(property, cluster, namespace);
-        validateReplicationSettingsOnNamespace(pulsar(), namespaceName);
-    }
-
     /**
      * If the namespace is global, validate the following - 1. If replicated clusters are configured for this global
      * namespace 2. If local cluster belonging to this namespace is replicated 3. If replication is enabled for this
-     * namespace
+     * namespace <br/>
+     * It validates if local cluster is part of replication-cluster. If local cluster is not part of the replication
+     * cluster then it redirects request to peer-cluster if any of the peer-cluster is part of replication-cluster of
+     * this namespace. If none of the cluster is part of the replication cluster then it fails the validation.
      *
-     * @param pulsarService
      * @param namespace
      * @throws Exception
      */
-    protected static void validateReplicationSettingsOnNamespace(PulsarService pulsarService, NamespaceName namespace) {
+    protected void validateGlobalNamespaceOwnership(NamespaceName namespace) {
         try {
-            validateReplicationSettingsOnNamespaceAsync(pulsarService, namespace).get(cacheTimeOutInSec, SECONDS);
+            ClusterData peerClusterData = checkLocalOrGetPeerReplicationCluster(pulsar(), namespace)
+                    .get(cacheTimeOutInSec, SECONDS);
+            // if peer-cluster-data is present it means namespace is owned by that peer-cluster and request should be
+            // redirect to the peer-cluster
+            if (peerClusterData != null) {
+                URI redirect = getRedirectionUrl(peerClusterData);
```

Comment:
To avoid cyclic request between clusters due to misconfiguration or other issues, should the request indicate that it's a redirected request?

Reply:
we may not need it because it won't do multiple redirections. The cluster which receives the lookup-request for a global-namespace,  if that cluster is not part of the replication-clusters then it will find out peer which is part of replication-cluster and can own that global-namespace. So, cluster will only redirect to peer if it finds out that selected peer can own the namespace else this cluster will reject the request. 
So, peer which will receive redirected-lookup request will always own this namespace and will never do another redirection.